environmentlighting

dhruvadhia1 opened the pull request:
<!-- Instructions: https://github.com/google/model-viewer/blob/master/CONTRIBUTING.md#code-reviews -->
### Reference Issue
<!-- Example: Fixes #1234 -->

elalish commented on packages/modelviewer.dev/examples/staging-and-camera-control.html:
```diff
@@ -299,7 +299,52 @@ <h4>Event listeners can cooperate with camera-controls</h4>
+        </div>
+        <example-snippet stamp-to="demo-container-8" highlight-as="html">
+          <template>
+			  <model-viewer id="pan-demo" camera-controls src="../shared-assets/models/Astronaut.glb" environment-image="../assets/whipple_creek_regional_park_04_1k.hdr" 				alt="A 3D model of an astronaut"></model-viewer>
```
This needs a unique id to ensure it doesn't conflict the other demos. Likewise, below you should select using `#your_id` so that you select the correct example.

dhruvadhia1 replied:
i see, ok let me make those changes.

elalish commented on packages/modelviewer.dev/examples/staging-and-camera-control.html:
```diff
@@ -299,7 +299,52 @@ <h4>Event listeners can cooperate with camera-controls</h4>
+          <template>
+			  <model-viewer id="pan-demo" camera-controls src="../shared-assets/models/Astronaut.glb" environment-image="../assets/whipple_creek_regional_park_04_1k.hdr" 				alt="A 3D model of an astronaut"></model-viewer>
+			  <script>
+				  const modelViewer = document.querySelector("model-viewer");
```
It looks like your editor may be set up to use tabs instead of spaces. Can you fix the formatting to match our other examples? It's also important to start inside the `<template>` with zero indent, as any spaces will be applied in the visible portion of the example webpage.

dhruvadhia1 replied:
sure

elalish commented on packages/modelviewer.dev/examples/staging-and-camera-control.html:
```diff
@@ -299,7 +299,52 @@ <h4>Event listeners can cooperate with camera-controls</h4>
+				    modelViewer.jumpCameraToGoal();
+					requestAnimationFrame(() => updateOrbit());
+			  	    var isMouseDown = false;
+			  	    document.onmousedown = function() { isMouseDown = true  };
```
Since we have multiple examples on this page, you should instead add an event listener to this `modelViewer` element instead of the whole document.

elalish commented on packages/modelviewer.dev/examples/staging-and-camera-control.html:
```diff
@@ -299,7 +299,52 @@ <h4>Event listeners can cooperate with camera-controls</h4>
+	  
+				  const updateOrbit = () => {
+				    const orbit = modelViewer.getCameraOrbit();
+				    modelViewer.cameraOrbit = `${orbit.theta + 0.01}rad ${orbit.phi}rad ${orbit.radius}m`;
```
Let's remove this line for simplicity as it is independent of what you're demonstrating here.

dhruvadhia1 replied:
you mean this line. 				    `modelViewer.cameraOrbit = `${orbit.theta + 0.01}rad ${orbit.phi}rad ${orbit.radius}m`;`
?

elalish replied:
correct.

elalish commented on packages/modelviewer.dev/examples/staging-and-camera-control.html:
```diff
@@ -299,7 +299,52 @@ <h4>Event listeners can cooperate with camera-controls</h4>
+			  		  if(isMouseDown) { 
+			  	  	    if (e.shiftKey && e.which == 1) {
+
+			  				modelViewer.resetTurntableRotation(-1 * orbit.theta);
```
Have you tried serving this example locally (`npm run serve`) and testing it? I'm curious if this has the desired effect when you first rotate and then shift-rotate?

dhruvadhia1 replied:
nope, I wasnt sure at exactly what stage this gets tested. But I can do that locally as well and let you know how it goes. Previously I tested with older model viewer code and that had no effect. But I believe I was not using the right version of model viewer at that time as resetTurnTable() didn't take any parameters.

dhruvadhia1 replied:
So I made all the above changes and I realised I had to npm install first and when I ran npm run build. I am getting bunch of error. Not sure about them. 

@google/3dom: ../../node_modules/@types/jest/index.d.ts(45,15): error TS2451: Cannot redeclare block-scoped variable 'expect'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2514,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'beforeEach' must be of type 'Lifecycle', but here has type 'HookFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2532,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'afterEach' must be of type 'Lifecycle', but here has type 'HookFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2548,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'describe' must be of type 'Describe', but here has type 'SuiteFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2569,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'xdescribe' must be of type 'Describe', but here has type 'PendingSuiteFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2583,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'it' must be of type 'It', but here has type 'TestFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2597,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'test' must be of type 'It', but here has type 'TestFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2604,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'xit' must be of type 'It', but here has type 'PendingTestFunction'.
@google/3dom: src/end-to-end-spec.ts(16,20): error TS2307: Cannot find module 'three/examples/jsm/loaders/GLTFLoader' or its corresponding type declarations.
@google/3dom: src/end-to-end-spec.ts(17,24): error TS2307: Cannot find module 'three/src/core/Object3D.js' or its corresponding type declarations.
@google/3dom: src/end-to-end-spec.ts(18,36): error TS2307: Cannot find module 'three/src/materials/MeshStandardMaterial.js' or its corresponding type declarations.
@google/3dom: src/end-to-end-spec.ts(19,20): error TS2307: Cannot find module 'three/src/objects/Mesh.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph-spec.ts(16,59): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph-spec.ts(17,35): error TS2307: Cannot find module 'three/examples/jsm/loaders/GLTFLoader.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph-spec.ts(18,29): error TS2307: Cannot find module 'three/examples/jsm/utils/SkeletonUtils.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(1,56): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(2,48): error TS2307: Cannot find module 'three/examples/jsm/loaders/GLTFLoader.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(70,27): error TS7006: Parameter 'gltfElementReference' implicitly has an 'any' type.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(70,49): error TS7006: Parameter 'threeObject' implicitly has an 'any' type.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(82,28): error TS7053: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type 'GLTF'.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(129,36): error TS7053: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type 'GLTF'.
@google/3dom: src/facade/three-js/image.ts(16,52): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/material-spec.ts(16,36): error TS2307: Cannot find module 'three/src/materials/MeshStandardMaterial.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/material-spec.ts(17,21): error TS2307: Cannot find module 'three/src/math/Color.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/material.ts(16,61): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/model-graft-spec.ts(16,24): error TS2307: Cannot find module 'three/src/core/Object3D.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/model-spec.ts(16,36): error TS2307: Cannot find module 'three/src/materials/MeshStandardMaterial.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/model-spec.ts(17,20): error TS2307: Cannot find module 'three/src/objects/Mesh.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/model-spec.ts(32,33): error TS7006: Parameter 'object' implicitly has an 'any' type.
@google/3dom: src/facade/three-js/model.ts(16,36): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/pbr-metallic-roughness-spec.ts(16,36): error TS2307: Cannot find module 'three/src/materials/MeshStandardMaterial.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/pbr-metallic-roughness-spec.ts(17,21): error TS2307: Cannot find module 'three/src/math/Color.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/pbr-metallic-roughness.ts(16,61): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/sampler-spec.ts(16,23): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/sampler.ts(16,39): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/texture-info.ts(16,39): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/texture-spec.ts(16,39): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/texture.ts(16,39): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/three-dom-element-spec.ts(17,24): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/three-dom-element-spec.ts(18,24): error TS2307: Cannot find module 'three/src/core/Object3D.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/three-dom-element.ts(16,43): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/test-helpers.ts(16,38): error TS2307: Cannot find module 'three' or its corresponding type declarations.

here is a log.

elalish replied:
Did you follow the README? It looks like you may have missed `npm run bootstrap`? Also, what OS are you on?

dhruvadhia1 replied:
Oh snaps, you are right, running that command, will let you know. I am on macOS catalina.

elalish reviewed (changes requested):
Thank you! A few comments, and make sure to sign the CLA or I can't merge your PR. It's just releasing your code into our Apache license.

elalish commented:
Any luck with getting it to build? And are you having any trouble with the CLA bot? Let me know when you'd like me to review again.

dhruvadhia1 commented:
> Any luck with getting it to build? And are you having any trouble with the CLA bot? Let me know when you'd like me to review again.

@elalish I got it all setup on my local machine and testing it now. I didn't get the desired result. Ill be spending sometime on it today.

dhruvadhia1 commented:
> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for all the commit author(s) or Co-authors. If you authored these, maybe you used a different email address in the git commits than was used to sign the CLA ([login here](https://cla.developers.google.com/) to double check)? If these were authored by someone else, then they will need to sign a CLA as well, and confirm that they're okay with these being contributed to Google.
> In order to pass this check, please resolve this problem and then comment `@googlebot I fixed it.`. If the bot doesn't comment, it means it doesn't think anything has changed.
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fgoogle%2Fmodel-viewer%2Fpull%2F1551) for more info**.
@googlebot I fixed it

dhruvadhia1 commented:
@elalish I dont have the output required, can you still review the code and see whats happening? On shift press and mouse move I see jittered movement, it seems like reset is trying out cancel out the rotation, but there is still some.

dhruvadhia1 commented:
@elalish not sure if you got a chance to look at last commit? I have tried few different approaches, but none seem to be working so far the way we discussed before based on the code I shared.

elalish commented:
@dhruvadhia1 If you sign the CLA (follow the google-cla bot's directions) I'll take a look. Otherwise I can't merge your code even if I fix it.

dhruvadhia1 commented:
![Screen Shot 2020-09-28 at 9 58 45 AM](https://user-images.githubusercontent.com/70779749/94463034-35c57f80-0171-11eb-9aea-d146bcc5a188.png)

I have done it already, and did that again through employer.

elalish commented:
@dhruvadhia1 As it says above, you also have to write `@googlebot I fixed it` in order to get confirmation.

elalish commented:
Perhaps you don't have your git email set to the one you signed with?

dhruvadhia1 commented:
@googlebot I fixed it

dhruvadhia1 commented:
> Perhaps you don't have your git email set to the one you signed with?

hmm. Its all through the company address dhruv@thecela.com both github account and account for contributing.

dhruvadhia1 commented:
> @dhruvadhia1 As it says above, you also have to write `@googlebot I fixed it` in order to get confirmation.

I have done that twice but it still cant find the agreement..

elalish commented:
@dhruvadhia1 Okay, it found the CLA for your github username, but it thinks some of the commits are associated with a strange email address instead: dhruvadhia@Dhruvs-MacBook-Pro.local

Maybe due to an odd git setting? When this happens they recommend rebasing with your email correctly set to match your github user. Sorry for the run-around... this is what happens when legal meets technology.

dhruvadhia1 commented:
> @dhruvadhia1 Okay, it found the CLA for your github username, but it thinks some of the commits are associated with a strange email address instead: [dhruvadhia@Dhruvs-MacBook-Pro.local](mailto:dhruvadhia@Dhruvs-MacBook-Pro.local)
> 
> Maybe due to an odd git setting? When this happens they recommend rebasing with your email correctly set to match your github user. Sorry for the run-around... this is what happens when legal meets technology.

got it and no worries. its just not that user friendly.

elalish commented:
Can you try `git config user.email` on your command line? That's the only thing I can think of that could be confusing it.

dhruvadhia1 commented:
@googlebot I fixed it

dhruvadhia1 commented:
@googlebot I fixed it

dhruvadhia1 commented:
> Can you try `git config user.email` on your command line? That's the only thing I can think of that could be confusing it.

git config user.email dhruv@thecela.com and pushed again. i can see the change in email when I do git log. But still same issue.

elalish commented:
I'm hoping that if you rebase again now that your email is fixed it'll finally work.

dhruvadhia-cela commented:
@googlebot I signed it!

dhruvadhia-cela commented:
> I'm hoping that if you rebase again now that your email is fixed it'll finally work.

I got it working with an entirely new account which is not a company account, just gmail. It seems the CLA has some issues with gsuite accounts? anyways, I made a new PR. I am not sure how to close this one. Here is the link to new one. https://github.com/google/model-viewer/pull/1570

elalish commented:
Awesome, thank you!
